dialog/fileitem.go

```diff
@@ -77,6 +96,11 @@ func (f *fileDialog) newFileItem(location fyne.URI, dir, up bool) *fileDialogIte
 	return item
 }
 
+func (i *fileDialogItem) setChooseAndOpenCallBack(choose func(id int), open func()) {
```

Comment:
A minor style thing - I'm not sure it's worth creating a function for just this. Instead what you could do is where you invoke this function now, do a cast to (*fileDialogItem) and save it to a local variable, then you can simply access .choose and .open (and .setLocation).